htdocs/product/class/api_products.class.php

```diff
@@ -219,9 +220,13 @@ public function index($sortfield = "t.ref", $sortorder = 'ASC', $limit = 100, $p
 			$i = 0;
 			while ($i < $min) {
 				$obj = $this->db->fetch_object($result);
-				$product_static = new Product($this->db);
-				if ($product_static->fetch($obj->rowid)) {
-					$obj_ret[] = $this->_cleanObjectDatas($product_static);
+				if (!$ids_only) {
+					$product_static = new Product($this->db);
+					if ($product_static->fetch($obj->rowid)) {
+						$obj_ret[] = $this->_cleanObjectDatas($product_static);
+					}
+				} else {
+					$obj_ret[] = $obj;
```

Comment:
@c3do is it $obj->rowid instead only $obj ?

Reply:
@frederic34 as you wish :)